crac/src/test/groovy/io/micronaut/crac/CracRegistrationSpec.groovy

```diff
@@ -20,11 +20,12 @@ class CracRegistrationSpec extends Specification {
     @Inject
     CracContextProviderReplacement cracContextProviderReplacement
 
-    def "resources are registered in the expected order"() {
+    void "resources are registered in the expected order"() {
         expect:
         cracContextProviderReplacement.replacement.registrations == [
                 'TestResource',
-                'NettyEmbeddedServerCracHander',
+                'RefreshEventResource',
```

Comment:
I think  refresh event should be last to allow other resources to execute before destroying beans

Reply:
You're probably right...

There's an issue though, as if the refresh is done after netty, we get

```
Error instantiating bean of type  [io.micronaut.web.router.DefaultRouter]

Message: Bean of type [class io.micronaut.context.DefaultApplicationContext$RuntimeConfiguredEnvironment] is a manually registered singleton that was registered with the context via BeanContext.registerBean(..) and cannot be created directly
Path Taken: NettyEmbeddedServer.buildDefaultServer(NettyHttpServerConfiguration configuration) --> new DefaultNettyEmbeddedServerFactory(ApplicationContext applicationContext,[RouteExecutor routeExecutor],MediaTypeCodecRegistry mediaTypeCodecRegistry,StaticResourceResolver staticResourceResolver,ThreadFactory nettyThreadFactory,HttpCompressionStrategy httpCompressionStrategy,EventLoopGroupFactory eventLoopGroupFactory,EventLoopGroupRegistry eventLoopGroupRegistry) --> new RouteExecutor([Router router],BeanContext beanContext,RequestArgumentSatisfier requestArgumentSatisfier,HttpServerConfiguration serverConfiguration,ErrorResponseProcessor errorResponseProcessor,ExecutorSelector executorSelector) --> new DefaultRouter([Collection builders])
io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.web.router.DefaultRouter]
```

Investigating...

Reply:
The test was at fault I believe.

Fixed in https://github.com/micronaut-projects/micronaut-crac/pull/18/commits/239a55ebe5d645f7c65c47a87d603d85fc1ad266